sycl/tools/syclbin-dump/syclbin-dump.cpp

```diff
@@ -0,0 +1,193 @@
+//==----------- syclbin-dump.cpp -------------------------------------------==//
```

Comment:
Joseph extended llvm-objdump for offloading scenario here:
https://github.com/llvm/llvm-project/commit/d2d8b0aa4f80752722321d1e74e2d98761caaa61

We could use this approach and allow users and us to print SYCLBIN's content by llvm-objdump tool.

Reply:
Oh, interesting! I'll have to dig deeper into how equivalent they are, but it sounds like the right path. My biggest worry is that this will complicate pulldowns even more for now, so I see two options:

1. Run with syclbin-dump as an initial tool for this. It lets us do introspective testing of the SYCLBIN format without affecting pulldown. When upstreaming, we transition to llvm-objdump.
2. Transition to llvm-objdump immediately, delaying this patch a little (or maybe doing it as a follow-up replacing syclbin-dump as it isn't super crucial while SYCLBIN is still being integrated) and living with the potential pulldown conflicts while it gets upstreamed.

@maksimsab @AlexeySachkov @asudarsa - Preferences?